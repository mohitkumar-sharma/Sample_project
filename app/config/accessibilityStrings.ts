export default {
  NORMAL_BUTTON_TEXT: 'normal-button-text',
  SLIDER_BUTTON_TEXT: 'slider-button-text',
  NORMAL_BUTTON: 'normal-button',
  SLIDER_BUTTON_ICON: 'slider-button-icon',
  SLIDER_BUTTON_CONTAINER: 'slider-button-container',
  BANNER_TEXT: 'banner-text',
  HEADER_SIDES_TEXT: 'header-sides-text',
  HEADER_PRESSABLE_ELEMENT: 'header-pressable-element',
  SETTINGS_SWITCH_TEXT: 'settings-switch-text',
  SETTINGS_SWITCH: 'settings-switch',
  INV_TEXT_INPUT: 'inv-text-input',
};
