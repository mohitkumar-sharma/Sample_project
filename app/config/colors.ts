export default {
  black: '#000000',
  gray: '#9B9B9B',
  divider: '#E8E8E8',
  white: '#FFFFFF',
  themeBlue: '#6EB1F7',
  themeLightBlack: '#34424A',
  appTheme: '#1A1924',
  borderLine: '#59606A',
  warning: '#FFCC00',
  red: '#FF0000',
};
