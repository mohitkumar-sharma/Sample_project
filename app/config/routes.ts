export default {
  WELCOME: 'Welcome',
  HOME: 'Home',
  SETTINGS: 'Settings',
};
