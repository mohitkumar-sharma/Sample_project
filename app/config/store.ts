import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';
import { rootReducer } from '../states';

const store = configureStore({
  reducer: rootReducer,
  middleware: getDefaultMiddleware(),
});

export type AppDispatch = typeof store.dispatch;

export { store };
