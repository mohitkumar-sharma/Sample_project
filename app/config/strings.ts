export default {
  GO_TO_HOME: 'Go To Home',
  DIAMOND_ICON: 'diamond',
  SLIDE_ME: 'Slide me to continue',
  APP_RUNNING_WARNING: 'Notice: You are running this app on',
  WELCOME_INPUT_PLACEHOLDER: 'Enter your name here...',
  SIGN_OUT: 'SignOut',
  HOME_BUTTONS_HEADING: '4 variations of a button',
  PRESS_ME: 'Press me',
  TRANSPARENT_BTN_PRESSED: 'Transparent button is pressed',
  LIGHT_BLACK_BTN_PRESSED: 'Light black button is pressed',
  BLUE_BTN_PRESSED: 'Blue button is pressed',
  SWITCH_TEXT: 'Enable Warning Banner',
  ALERT_TITLE_INVALID_USER_NAME: 'Invalid UserName!',
  ALERT_TITLE_BTN_PRESSED: 'Button Pressed!',
  ALERT_ENTER_VALID_USER_NAME: 'Please enter a valid user name.',
};
