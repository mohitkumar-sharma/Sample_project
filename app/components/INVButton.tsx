import React from 'react';
import { TouchableOpacity, Text, StyleSheet, StyleProp, ViewStyle, TextStyle } from 'react-native';
import config from '../config';

export interface INVButtonProps {
  /* as an string to show the text of button on Text element */
  buttonText?: string;
  /* as an object of style props for TouchableOpacity element */
  containerStyle?: StyleProp<ViewStyle>;
  /* as an object of style props for Text element */
  buttonTextStyle?: StyleProp<TextStyle>;
  /* as a function triggered when pressed */
  onPress?: () => void;
}

/**
 * This is the INVButton functional component works as a button and can be customized from where it is used
 * @param props : props are defined in INVButtonProps which can be passed to this component
 */
const INVButton = (props: INVButtonProps): JSX.Element => {
  const { buttonText, containerStyle, buttonTextStyle, onPress } = props;
  return (
    <TouchableOpacity
      style={[styles.container, containerStyle]}
      onPress={onPress}
      disabled={!onPress}
      testID={config.accessibilityStrings.NORMAL_BUTTON}>
      {buttonText && buttonText !== '' ? (
        <Text
          style={[styles.buttonText, buttonTextStyle]}
          numberOfLines={1}
          testID={config.accessibilityStrings.NORMAL_BUTTON_TEXT}>
          {buttonText}
        </Text>
      ) : null}
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  container: {
    backgroundColor: config.colors.themeBlue,
    height: 55,
    marginHorizontal: 20,
    borderRadius: 8,
    justifyContent: 'center',
  },
  buttonText: {
    fontSize: 16,
    fontWeight: config.fontWeights.FONT_WEIGHT_REGULAR,
    color: config.colors.white,
    textAlign: 'center',
  },
});

export default INVButton;
