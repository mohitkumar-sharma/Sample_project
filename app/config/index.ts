import colors from './colors';
import localConfig from './localConfig';
import strings from './strings';
import routes from './routes';
import fontWeights from './fontWeights';
import accessibilityStrings from './accessibilityStrings';
import { store } from './store';

const config = {
  colors,
  ...localConfig,
  store,
  strings,
  routes,
  fontWeights,
  accessibilityStrings,
};

export default config;
