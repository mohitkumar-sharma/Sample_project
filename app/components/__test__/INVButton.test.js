import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import INVButton from '../INVButton';
import config from '../../config';

describe('INVButton', () => {
  const props = {
    containerStyle: { backgroundColor: config.colors.appTheme },
    buttonTextStyle: { backgroundColor: config.colors.themeBlue, fontSize: 17 },
  };

  const buttonText = config.strings.GO_TO_HOME;
  const onPress = jest.fn();

  it('matches snapshot', () => {
    expect(render(<INVButton {...props} />)).toMatchSnapshot();
    expect(
      render(<INVButton {...props} buttonText={buttonText} onPress={onPress} />)
    ).toMatchSnapshot();
  });

  it('renders the button text', () => {
    const { queryByTestId } = render(<INVButton {...props} buttonText={buttonText} />);
    expect(queryByTestId(config.accessibilityStrings.NORMAL_BUTTON_TEXT)).toBeTruthy();
  });

  it('does not render the button text', () => {
    const { queryByTestId } = render(<INVButton {...props} />);
    expect(queryByTestId(config.accessibilityStrings.NORMAL_BUTTON_TEXT)).toBeFalsy();
  });

  it('button press', () => {
    const buttonPress = jest.fn();
    const { getByTestId } = render(
      <INVButton {...props} buttonText={buttonText} onPress={buttonPress} />
    );

    const button = getByTestId(config.accessibilityStrings.NORMAL_BUTTON);
    fireEvent.press(button);

    expect(buttonPress).toHaveBeenCalledWith();
  });
});
