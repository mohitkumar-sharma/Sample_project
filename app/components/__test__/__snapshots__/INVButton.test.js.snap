// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`INVButton matches snapshot 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#1A1924",
      "borderRadius": 8,
      "height": 55,
      "justifyContent": "center",
      "marginHorizontal": 20,
      "opacity": 1,
    }
  }
  testID="normal-button"
/>
`;

exports[`INVButton matches snapshot 2`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#1A1924",
      "borderRadius": 8,
      "height": 55,
      "justifyContent": "center",
      "marginHorizontal": 20,
      "opacity": 1,
    }
  }
  testID="normal-button"
>
  <Text
    numberOfLines={1}
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontSize": 16,
          "fontWeight": "400",
          "textAlign": "center",
        },
        Object {
          "backgroundColor": "#6EB1F7",
          "fontSize": 17,
        },
      ]
    }
    testID="normal-button-text"
  >
    Go To Home
  </Text>
</View>
`;
